lib/index.ts

```diff
@@ -4,14 +4,16 @@ export { isValidationErrorLike } from './isValidationErrorLike.ts';
 export { isZodErrorLike } from './isZodErrorLike.ts';
 export { errorMap } from './errorMap.ts';
 export { fromError } from './fromError.ts';
-export {
-  fromZodIssue,
-  type FromZodIssueOptions,
-  type ZodIssue,
-} from './fromZodIssue.ts';
+export { fromZodIssue, type FromZodIssueOptions } from './fromZodIssue.ts';
 export {
   fromZodError,
   type FromZodErrorOptions,
   type ZodError,
 } from './fromZodError.ts';
 export { toValidationError } from './toValidationError.ts';
+export {
+  type MessageBuilder,
+  type ZodIssue,
+  createMessageBuilder,
+} from './MessageBuilder.ts';
+export { type NonEmptyArray } from './utils/NonEmptyArray.ts';
```

Comment:
[question] How come we export this util too? Are we expecting the consumers of the lib to use it?

Reply:
Yes, because they will need to create their own `messageBuilder` implementations, right? This is the point of the PR. So, they need the `NonEmptyArray` definition.